app/javascript/src/utilities/ReCaptchaV3.tsx

```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react'
+
+import { useScript } from 'utilities/useScript'
+
+import type { ReactElement } from 'react'
+
+interface Props {
+  enabled?: boolean;
+  siteKey: string;
+  action: string;
+}
+
+const ReCaptchaV3 = ({ enabled = false, siteKey, action }: Props): ReactElement => {
+  if (!enabled) return (<div>{null}</div>)
+
+  const inputId = `g-recaptcha-response-data-${action}`.replace(/\//g, '-')
+  const inputName = `g-recaptcha-response-data[${action}]`
+  const [token, setToken] = useState('')
+
+  useScript(`https://www.google.com/recaptcha/api.js?render=${siteKey}`, () => {
```

Comment:
Why do you need to useEffect this? Is `siteKey` ever going to change dynamically? Maybe I am mistaken but to me it looks like `siteKey` is passed once to the component, at first render. If so, there's no need to use an effect, simply render the script.

I wonder if it would make sense to do it in the template (new.html.slim) and use a defer:
```slim
<script id="recaptcha_script" src="https://www.google.com/recaptcha/api.js?render=#{siteKey}" defer />
```

`defer` will load the script but wait for the HTML to be parsed before executing the script, but before DOMContentLoaded is triggered.

Reply:
I did it because...

![I have no idea what I'm doing](https://i0.kym-cdn.com/entries/icons/original/000/008/342/ihave.jpg)

I think I'm going to keep the `useScript` component but remove the call to `useEffect`. I think that's better than just rendering `<script />` because this way I can run he callback after the script has been loaded. I'm not sure it would work the same way with `defer`